### PR TITLE
Include revert strings in IR

### LINKED ISSUE #8252: Add internal revert reasons to Yul codegen
Similar to https://github.com/ethereum/solidity/pull/8178

- [ ] `codegen/YulUtilFunctions.cpp`
- [ ] `codegen/ir/*`
- [ ] tests
- [ ] docs

### PR DESCRIPTION
**Pull request by axic**
Fixes #8252

### CODE REVIEW

**Comment by chriseth** [comment]
These are not the only revert strings missing, are they?

**Comment by mijovic** [comment]
> These are not the only revert strings missing, are they?

These are only once that differ with old codegen, at least what was discovered with tests.

**Comment by mijovic** [comment]
> > These are not the only revert strings missing, are they?
> 
> These are only once that differ with old codegen, at least what was discovered with tests.

File: libsolidity/codegen/ir/IRGenerator.cpp [hunk 1]
```diff
@@ -818,7 +818,7 @@ string IRGenerator::dispatchRoutine(ContractDefinition const& _contract)
 				return(memPos, sub(memEnd, memPos))
 			}
 			</cases>
-			default {}
+			default { }
```
Comment (axic) [inline libsolidity/codegen/ir/IRGenerator.cpp]: Why is this change needed?

**Comment by axic** [comment]
Looks good to me anyhow.

File: libsolidity/codegen/ir/IRGenerator.cpp (hunk 1 above)
Comment (mijovic) [reply libsolidity/codegen/ir/IRGenerator.cpp]: Not needed, included by mistake, removing it now

File: libsolidity/codegen/ir/IRGenerator.cpp [hunk 2]
```diff
@@ -839,8 +839,10 @@ string IRGenerator::dispatchRoutine(ContractDefinition const& _contract)
 			if (type->stateMutability() > StateMutability::View)
 				// If the function is not a view function and is called without DELEGATECALL,
 				// we revert.
-				// TODO add revert message.
-				delegatecallCheck = "if iszero(called_via_delegatecall) { revert(0, 0) }";
+				delegatecallCheck =
+					"if iszero(called_via_delegatecall) {" +
+					m_context.revertReasonIfDebug("Non-view function of library called without DELEGATECALL") +
+					" }";
```
Comment (chriseth) [inline libsolidity/codegen/ir/IRGenerator.cpp]:
```suggestion
					"if iszero(called_via_delegatecall) { " +
					m_context.revertReasonIfDebug("Non-view function of library called without DELEGATECALL") +
					" }";
```

**Review by chriseth — APPROVED** [review]
